src/Jackalope/ImportExport/ImportExport.php

```diff
@@ -464,10 +502,10 @@ private static function importSystemView(NodeInterface $parentNode, NamespaceReg
         if (! $xml->read()) {
             throw new InvalidSerializedDataException('missing information to create node');
         }
-        if ('property' != $xml->localName || NamespaceRegistryInterface::NAMESPACE_SV != $xml->namespaceURI) {
+        if ('property' !== $xml->localName || NamespaceRegistryInterface::NAMESPACE_SV !== $xml->namespaceURI) {
             throw new InvalidSerializedDataException('first child of node must be sv:property for jcr:primaryType. Found {'.$xml->namespaceURI.'}'.$xml->localName.'="'.$xml->value.'"'.$xml->nodeType);
         }
-        if (! $xml->moveToAttributeNs('name', NamespaceRegistryInterface::NAMESPACE_SV) || 'jcr:primaryType' != $xml->value) {
+        if ('jcr:primaryType' !== $xml->value || !$xml->moveToAttributeNs('name', NamespaceRegistryInterface::NAMESPACE_SV)) {
```

Comment:
i tested with jackalope-doctrine-dbal, and this line is a regression. the problem is that we call moveToAttributeNs for its side effect of moving. plus, the check for value must happen after we moved to the attribute, not before.

to make this more explicit, we could split this in two like this:

```
if (! $xml->moveToAttributeNs('name', NamespaceRegistryInterface::NAMESPACE_SV)) {
    throw new InvalidSerializedDataException('first child of node must have a sv:name property');
}
if ('jcr:primaryType' !== $xml->value) {
    throw new InvalidSerializedDataException('first child of node must be the sv:property node with a jcr:primaryType. Found {'.$xml->namespaceURI.'}'.$xml->localName.'="'.$xml->value
}
```

do you want to update like this? i think if you change that, this would be ready to merge.

(if you want, you could search this file for " == " - i found some places where we don't use === that you missed.